### PR TITLE
Add `aws_iam_open_id_connect_provider` table ( #1802)

### PR DESCRIPTION
**Pull request by LalitLab**
# Integration test logs
<details>
  <summary>Logs</summary>

```
➜  aws-test git:(1802-add-iam-openidproviders-table) ./tint.js tests/aws_iam_open_id_connect_provider 
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/aws_iam_open_id_connect_provider []

PRETEST: tests/aws_iam_open_id_connect_provider

TEST: tests/aws_iam_open_id_connect_provider
Running terraform
data.aws_region.alternate: Reading...
data.aws_region.alternate: Read complete after 0s [id=us-east-2]
data.aws_partition.current: Reading...
data.aws_region.primary: Reading...
data.aws_caller_identity.current: Reading...
data.aws_region.primary: Read complete after 0s [id=us-east-1]
data.aws_partition.current: Read complete after 0s [id=aws]
data.aws_caller_identity.current: Read complete after 1s [id=111122223333]

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # aws_iam_openid_connect_provider.named_test_resource will be created
  + resource "aws_iam_openid_connect_provider" "named_test_resource" {
      + arn             = (known after apply)
      + client_id_list  = [
          + "266362248691-342342xasdasdasda-apps.googleusercontent.com",
        ]
      + id              = (known after apply)
      + tags            = {
          + "name" = "turbottest79877"
        }
      + tags_all        = {
          + "name" = "turbottest79877"
        }
      + thumbprint_list = [
          + "cf23df2207d99a74fbe169e3eba035e633b65d94",
          + "6938fd4d98bab03faadb97b34396831e3780aea1",
        ]
      + url             = "https://accounts.google.com"
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + account_id    = "111122223333"
  + aws_partition = "aws"
  + aws_region    = "global"
  + resource_aka  = (known after apply)
  + resource_name = "turbottest79877"
aws_iam_openid_connect_provider.named_test_resource: Creating...
aws_iam_openid_connect_provider.named_test_resource: Creation complete after 1s [id=arn:aws:iam::111122223333:oidc-provider/accounts.google.com]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

account_id = "111122223333"
aws_partition = "aws"
aws_region = "global"
resource_aka = "arn:aws:iam::111122223333:oidc-provider/accounts.google.com"
resource_name = "turbottest79877"

Running SQL query: test-get-query.sql
[
  {
    "arn": "arn:aws:iam::111122223333:oidc-provider/accounts.google.com",
    "region": "global",
    "tags": {
      "name": "turbottest79877"
    }
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "arn": "arn:aws:iam::111122223333:oidc-provider/accounts.google.com",
    "region": "global",
    "tags": {
      "name": "turbottest79877"
    }
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "arn": "arn:aws:iam::111122223333:oidc-provider/accounts.google.com",
    "region": "global",
    "tags": {
      "name": "turbottest79877"
    }
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-fields-query.sql
[
  {
    "akas": [
      "arn:aws:iam::111122223333:oidc-provider/accounts.google.com"
    ],
    "region": "global",
    "tags": {
      "name": "turbottest79877"
    }
  }
]
✔ PASSED

POSTTEST: tests/aws_iam_open_id_connect_provider

TEARDOWN: tests/aws_iam_open_id_connect_provider

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>

```sql
select
  arn,
  create_date,
  client_id_list,
  thumbprint_list,
  url,
  account_id
from
  aws_iam_open_id_connect_provider;
```
```
+-----------------------------------------------------------------------------+---------------------------+---------------------------------------------------------------+-----------------------------------------------------------------------------------------+-------------------------------------+--------------+
| arn                                                                         | create_date               | client_id_list                                                | thumbprint_list                                                                         | url                                 | account_id   |
+-----------------------------------------------------------------------------+---------------------------+---------------------------------------------------------------+-----------------------------------------------------------------------------------------+-------------------------------------+--------------+
| arn:aws:iam::111122223333:oidc-provider/accounts.google.com                 | 2023-07-01T12:30:02+05:30 | ["266362248691-342342xasdasdasda-apps.googleusercontent.com"] | ["cf23df2207d99a74fbe169e3eba035e633b65d94","6938fd4d98bab03faadb97b34396831e3780aea1"] | accounts.google.com                 | 111122223333 |
| arn:aws:iam::111122223333:oidc-provider/token.actions.githubusercontent.com | 2023-06-29T21:24:57+05:30 | ["sts.amazonaws.com"]                                         | ["1c58a3a8518e8759bf075b76b750d4f2df264fcd","6938fd4d98bab03faadb97b34396831e3780aea1"] | token.actions.githubusercontent.com | 111122223333 |
+-----------------------------------------------------------------------------+---------------------------+---------------------------------------------------------------+-----------------------------------------------------------------------------------------+-------------------------------------+--------------+
```

```sql
select
  arn,
  create_date,
  client_id_list,
  thumbprint_list,
  url,
  account_id
from
  aws_iam_open_id_connect_provider
where
  create_date <= (current_date - interval '90' day)
order by
  create_date;
```

```
+-----+-------------+----------------+-----------------+-----+------------+
| arn | create_date | client_id_list | thumbprint_list | url | account_id |
+-----+-------------+----------------+-----------------+-----+------------+
+-----+-------------+----------------+-----------------+-----+------------+
```

```sql
select
  arn,
  create_date,
  client_id_list,
  thumbprint_list,
  tags,
  url,
  account_id
from 
  aws_iam_open_id_connect_provider
where 
  tags ->> 'Timon' = 'Pumba';
```

```
+-----------------------------------------------------------------------------+---------------------------+-----------------------+-----------------------------------------------------------------------------------------+-------------------+-------------------------------------+--------------+
| arn                                                                         | create_date               | client_id_list        | thumbprint_list                                                                         | tags              | url                                 | account_id   |
+-----------------------------------------------------------------------------+---------------------------+-----------------------+-----------------------------------------------------------------------------------------+-------------------+-------------------------------------+--------------+
| arn:aws:iam::111122223333:oidc-provider/token.actions.githubusercontent.com | 2023-06-29T21:24:57+05:30 | ["sts.amazonaws.com"] | ["1c58a3a8518e8759bf075b76b750d4f2df264fcd","6938fd4d98bab03faadb97b34396831e3780aea1"] | {"Timon":"Pumba"} | token.actions.githubusercontent.com | 111122223333 |
+-----------------------------------------------------------------------------+---------------------------+-----------------------+-----------------------------------------------------------------------------------------+-------------------+-------------------------------------+--------------+
```

</details>

### CODE REVIEW

File: aws/table_aws_iam_open_id_connect_provider.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+		},
+		Columns: awsGlobalRegionColumns([]*plugin.Column{
+			{
+				Name:        "arn",
+				Description: "The Amazon Resource Name (ARN) specifying the OIDC provider resource.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "client_id_list",
+				Description: "A list of client IDs (also known as audiences) that are associated with the specified IAM OIDC provider resource object",
```
Comment (misraved) [inline aws/table_aws_iam_open_id_connect_provider.go:39]:
```suggestion
				Description: "A list of client IDs (also known as audiences) that are associated with the specified IAM OIDC provider resource object.",
```

File: aws/table_aws_iam_open_id_connect_provider.go [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+//// HYDRATE FUNCTIONS
+
+func getIamOpenIdConnectProvider(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	var arn string
+	if h.Item != nil {
+		provider := h.Item.(types.OpenIDConnectProviderListEntry)
+		arn = *provider.Arn
+	} else {
+		arn = d.EqualsQuals["arn"].GetStringValue()
```
Comment (misraved) [inline aws/table_aws_iam_open_id_connect_provider.go:130]:
```suggestion
		arn = d. EqualsQualString("arn")
```

File: docs/tables/aws_iam_open_id_connect_provider.md
Comment (misraved) [inline docs/tables/aws_iam_open_id_connect_provider.md:1]: @LalitLab could we add example queries using the `client_id_list` and `thumbprint_list` columns?

File: aws/table_aws_iam_open_id_connect_provider.go [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+		Name:        "aws_iam_open_id_connect_provider",
+		Description: "AWS IAM OpenID Connect Provider",
+		List: &plugin.ListConfig{
+			Hydrate: listIamOpenIdConnectProviders,
+		},
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.AllColumns([]string{"arn"}),
+			Hydrate:    getIamOpenIdConnectProvider,
+			IgnoreConfig: &plugin.IgnoreConfig{
+				ShouldIgnoreErrorFunc: shouldIgnoreErrors([]string{"NoSuchEntity"}),
```
Comment (misraved) [inline aws/table_aws_iam_open_id_connect_provider.go:28]:
```suggestion
				ShouldIgnoreErrorFunc: shouldIgnoreErrors([]string{"NoSuchEntity", "InvalidInput"}),
```

**Review by misraved — COMMENTED** [review]
Thanks @LalitLab for raising a PR for this new table.

I have left a few review comments. Please let us know if you have questions 👍.

File: aws/table_aws_iam_open_id_connect_provider.go [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+//// HYDRATE FUNCTIONS
+
+func getIamOpenIdConnectProvider(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	var arn string
+	if h.Item != nil {
+		provider := h.Item.(types.OpenIDConnectProviderListEntry)
+		arn = *provider.Arn
+	} else {
+		arn = d. EqualsQualString("arn")
```
Comment (ParthaI) [inline aws/table_aws_iam_open_id_connect_provider.go:130]:
```suggestion
		arn = d.EqualsQualString("arn")
```

File: docs/tables/aws_iam_open_id_connect_provider.md
Comment (LalitLab) [reply docs/tables/aws_iam_open_id_connect_provider.md:1]: @misraved 
I have added the example for `client_id_list` and `thumbprint_list`. Let me know if anything else needs to be done on this PR.
Thank you

#### List AWS OpenID Providers without required thumbprint for audience 'sts.amazonaws.com'
```sql
select
  arn,
  create_date,
  client_id_list,
  thumbprint_list,
  tags,
  url,
  account_id
from
  aws_iam_open_id_connect_provider
where
  client_id_list @> '["sts.amazonaws.com"]'::jsonb
  and NOT thumbprint_list @> '["1c58a3a8518e8759bf075b76b750d4f2df264fcd", "6938fd4d98bab03faadb97b34396831e3780aea1"]'::jsonb
```

File: docs/tables/aws_iam_open_id_connect_provider.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  thumbprint_list,
+  tags,
+  url,
+  account_id
+from
+  aws_iam_open_id_connect_provider
+where
+  client_id_list @> '["sts.amazonaws.com"]'::jsonb
+  and NOT thumbprint_list @> '["1c58a3a8518e8759bf075b76b750d4f2df264fcd", "6938fd4d98bab03faadb97b34396831e3780aea1"]'::jsonb
+```
```
Comment (misraved) [inline docs/tables/aws_iam_open_id_connect_provider.md:72]:
```suggestion
### List AWS OpenID Providers without the required thumbprint for audience 'sts.amazonaws.com'

```sql
select
  arn,
  create_date,
  client_id_list,
  thumbprint_list,
  tags,
  url,
  account_id
from
  aws_iam_open_id_connect_provider
where
  client_id_list @> '["sts.amazonaws.com"]'::jsonb
  and not thumbprint_list @> '["1c58a3a8518e8759bf075b76b750d4f2df264fcd", "6938fd4d98bab03faadb97b34396831e3780aea1"]'::jsonb
```
```

**Comment by LalitLab** [comment]
> Thanks @LalitLab for adding an insightful example query. I have left a minor suggestion, everything else looks great 👍.

Thank you @misraved 
I have committed your recommended suggestion on the example query.

**Review by misraved — APPROVED** [review]

**Pull request merged**